spec/controllers/openid_connect/authorization_controller_spec.rb

```diff
@@ -0,0 +1,133 @@
+require 'rails_helper'
+
+RSpec.describe OpenidConnect::AuthorizationController do
+  let(:client_id) { 'urn:gov:gsa:openidconnect:test' }
+  let(:params) do
+    {
+      acr_values: 'http://idmanagement.gov/ns/assurance/loa/1',
+      client_id: client_id,
+      nonce:  SecureRandom.hex,
+      prompt: 'select_account',
+      redirect_uri: 'gov.gsa.openidconnect.test://result',
+      response_type: 'code',
+      scope: 'openid profile',
+      state:  SecureRandom.hex
+    }
+  end
+
+  describe '#index' do
+    subject(:action) { get :index, params }
+
+    context 'user is signed in' do
+      before do
+        user = create(:user, :signed_up)
+        sign_in user
+      end
+
+      context 'with valid params' do
+        it 'renders the approve/deny form' do
+          action
+          expect(controller).to render_template('openid_connect/authorization/index')
+        end
+
+        it 'track the event' do
+          stub_analytics
+          expect(@analytics).to receive(:track_event).
+            with(Analytics::OPENID_CONNECT_REQUEST_AUTHORIZATION,
+                 success: true,
+                 client_id: client_id,
+                 errors: [])
+
+          action
+        end
+      end
+
+      context 'with invalid params' do
+        before { params.delete(:state) }
+
+        it 'renders the error page' do
+          action
+          expect(controller).to render_template('openid_connect/authorization/error')
+        end
+
+        it 'track the event with errors' do
+          stub_analytics
+          expect(@analytics).to receive(:track_event).
+            with(Analytics::OPENID_CONNECT_REQUEST_AUTHORIZATION,
+                 success: false,
+                 client_id: client_id,
+                 errors: ['State Please fill in this field.'])
```

Comment:
Instead of using `errors.full_messages`, should we use `errors.messages`? That way, we get a nice hash with the key representing the field that contains the error, and the value is the error message:
```ruby
errors: { state: ['Please fill in this field.'] }
```